ui/packages/components/src/DatePicker/timeSchemas.ts

```diff
@@ -47,3 +63,15 @@ export const periodSchema = z
   .refine((value) => value === 'am' || value === 'pm', {
     message: 'Period must be "AM" or "PM".',
   });
+
+export const yearSchema = z.coerce
+  .number()
+  .int()
+  .min(1970, { message: 'Year must be a 4 digit number' })
+  .max(3000, { message: 'Year must be a 4 digit number' });
+
+export const monthSchema = z.coerce
+  .number()
+  .int()
+  .min(1, { message: 'Month must be 1 - 12' })
```

Comment:
```suggestion
  .min(1, { message: 'Month must be between 1 and 12' })
```
It's slightly more verbose but it's better for screen readers